File: 백준/Bronze/9506. 약수들의 합/약수들의 합.cc
#include <stdio.h>
#include <iostream>


using namespace std;

int main(){
    int n;
    int arr[100001];
    while(1){
        int p = 0;
        int sum = 0;
        cin >> n;
        if(n == -1){
            break;
        }
        for(int i = 1; i < n; i++){
            if(n%i == 0){
                arr[p++] = i;
                sum += i;
            }
        }
        if(sum == n){
            cout << n << " = " ;
            for(int i = 0; i < p-1; i++){
                cout << arr[i] << " + ";
                arr[i] = 0;
            }
            cout << arr[p-1] << '\n';
            arr[p-1] = 0;
        }else{
            cout << n << " is NOT perfect." << '\n';
        }
    }
}

File: 백준/Bronze/9506. 약수들의 합/약수들의 합.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringBuilder sb = new StringBuilder();

        while(true) {
            int n = Integer.parseInt(br.readLine());
            if (n == -1) {
                break;
            }
            int sum = 1;
            String str = n + " = 1";
            for (int i = 2; i < n; i++) {
                if (n % i == 0) {
                    sum += i;
                    str += " + " + i;
                }
            }

            if (sum != n) {
                str = n + " is NOT perfect.";
            }
            sb.append(str).append("\n");

        }


        bw.write(sb.toString());
        bw.flush();
    }
}